standalone-metastore/metastore-server/src/main/java/org/apache/hadoop/hive/metastore/datasource/HikariCPDataSourceProvider.java

```diff
@@ -72,6 +72,17 @@ public DataSource create(Configuration hdpConfig, int maxPoolSize) throws SQLExc
       config.setPoolName(poolName);
     }
 
+    // It's kind of a waste to create a fixed size connection pool as same as the TxnHandler#connPool,
+    // TxnHandler#connPoolMutex is mostly used for MutexAPI that is primarily designed to
+    // provide coarse-grained mutex support to maintenance tasks running inside the Metastore,
+    // add minimumIdle=2 and idleTimeout=5min to the pool, so that the connection pool can retire
+    // the idle connection aggressively, this will make Metastore more scalable especially if
+    // there is a leader in the warehouse.
+    if ("mutex".equals(poolName)) {
+      config.setMinimumIdle(Math.min(maxPoolSize, 2));
```

Comment:
Should we have the same setup for DbCPDataSource?

Reply:
Done, and added a test: https://github.com/apache/hive/pull/3817/files#diff-21013298e701863201669c0a08dce28bff302bca654dda0287b76cb344ae2c7a

Reply:
For hikari, the idle connections would be evicted from the pool only when minimumIdle is defined to be less than maximumPoolSize, however for DbCP, the idle object evictor thread runs only when timeBetweenEvictionRunsMillis is positive:
https://github.com/apache/commons-pool/blob/master/src/main/java/org/apache/commons/pool2/impl/BaseGenericObjectPool.java#L1810-L1830
Fixed Hikari not to override the idle config if the user has already provided the `hikaricp.minimumIdle`.